sdk/core/core/src/az_json_parser.c

```diff
@@ -277,10 +318,10 @@ az_json_parser_parse_token(az_json_parser * self, az_json_token * out_token) {
   if (!az_json_parser_stack_is_empty(self)) {
     return AZ_ERROR_JSON_INVALID_STATE;
   }
-  az_span_reader * const p_reader = &self->_internal.reader;
-  az_span_reader_skip_json_white_space(p_reader);
+  az_span * p_reader = &self->_internal.reader;
+  AZ_RETURN_IF_FAILED(az_span_reader_skip_json_white_space(p_reader));
   AZ_RETURN_IF_FAILED(az_json_parser_get_value_space(self, out_token));
-  bool const is_empty = az_span_reader_is_empty(p_reader);
+  bool const is_empty = az_span_length(*p_reader) <= 0; // everything was read
```

Comment:
I think I'd get rid of <0; a span should never be negative. If it is, we have a serious bug somewhere, let's find it quickly.